### PR TITLE
update email for Rey Lejano in groups/sig-release/groups.yaml

### PR DESCRIPTION
**Pull request by reylejano**
Update email for Rey Lejano

### CODE REVIEW

File: groups/sig-release/groups.yaml [hunk 1]
```diff
@@ -400,7 +400,7 @@ groups:
       - monmonmsc@gmail.com # 1.20 Bug Triage Shadow
       - pal.nabarun95@gmail.com  # 1.20 RT Lead Shadow
       - prashsh1@in.ibm.com # 1.20 CI Signal Shadow
-      - rey.lejano@rx-m.com # 1.20 Docs Shadow
+      - rlejano@gmail.com # 1.21 Docs Lead
```
Comment (palnabarun) [inline groups/sig-release/groups.yaml:403]: @reylejano -- Can you please remove your email ID from here? This alias is specifically for shadows.

**Review by palnabarun — CHANGES_REQUESTED** [review]

File: groups/sig-release/groups.yaml (hunk 1 above)
Comment (reylejano) [reply groups/sig-release/groups.yaml]: Removed email ID and squashed commits

**Review by cpanato — APPROVED** [review]
/lgtm

**Pull request merged**